controllers/chat/moveToAnon.js

```diff
@@ -86,6 +86,11 @@ const moveToAnon = async (req, res) => {
           newStateMemberWithAnonInfo[member].anon_user_info_emoji_code = anon_user_info_emoji_code;
           newStateMemberWithAnonInfo[member].anon_user_info_emoji_name = anon_user_info_emoji_name;
 
+          newStateMemberWithAnonInfo[member].user.name =
+            anon_user_info_color_name + ' ' + anon_user_info_emoji_name;
+          newStateMemberWithAnonInfo[member].user.username =
+            anon_user_info_color_name + ' ' + anon_user_info_emoji_name;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is directly modifying the `user` object's `name` and `username` properties. This could potentially lead to unintended side effects if other parts of the application are using these properties. Consider creating a new object with the updated properties instead of mutating the original one.

```diff
-           newStateMemberWithAnonInfo[member].user.name =
-             anon_user_info_color_name + ' ' + anon_user_info_emoji_name;
-           newStateMemberWithAnonInfo[member].user.username =
-             anon_user_info_color_name + ' ' + anon_user_info_emoji_name;

+           newStateMemberWithAnonInfo[member].user = {
+             ...newStateMemberWithAnonInfo[member].user,
+             name: anon_user_info_color_name + ' ' + anon_user_info_emoji_name,
+             username: anon_user_info_color_name + ' ' + anon_user_info_emoji_name,
+           };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->